### PR TITLE
CLOUDSTACK-9606: While IP address is released, tag are not deleted.

### PR DESCRIPTION
**Pull request by priyankparihar**
For complete description please refer -> [CLOUDSTACK-9606](https://issues.apache.org/jira/browse/CLOUDSTACK-9606)

### CODE REVIEW

**Comment by syed** [comment]
can you please provide a description and steps to reproduce

File: server/src/com/cloud/network/IpAddressManagerImpl.java [hunk 1]
```diff
@@ -168,6 +168,9 @@
 import com.cloud.vm.dao.NicSecondaryIpDao;
 import com.cloud.vm.dao.UserVmDao;
 import com.cloud.vm.dao.VMInstanceDao;
+import com.cloud.server.ResourceTag;
+import com.cloud.tags.dao.ResourceTagDao;
+
```
Comment (rajesh-battala) [inline server/src/com/cloud/network/IpAddressManagerImpl.java]: Remove this extra line

File: server/src/com/cloud/network/IpAddressManagerImpl.java [hunk 2]
```diff
@@ -568,6 +573,11 @@ protected boolean cleanupIpResources(long ipId, long userId, Account caller) {
             success = false;
         }
 
+        //Remove the tags corresponding to IP.
+        if(success) {
+            _resourceTagDao.removeByIdAndType(ipId, ResourceTag.ResourceObjectType.PublicIpAddress);
+        }
+
```
Comment (rajesh-battala) [inline server/src/com/cloud/network/IpAddressManagerImpl.java]: This extra line can be removed

File: server/src/com/cloud/network/IpAddressManagerImpl.java [hunk 3]
```diff
@@ -281,6 +284,8 @@
     VpcDao _vpcDao;
     SearchBuilder<IPAddressVO> AssignIpAddressSearch;
     SearchBuilder<IPAddressVO> AssignIpAddressFromPodVlanSearch;
+    @Inject
+    ResourceTagDao _resourceTagDao;
```
Comment (rajesh-battala) [inline server/src/com/cloud/network/IpAddressManagerImpl.java]: Move 287,288 after 284.

**Comment by priyankparihar** [comment]
Hi @syed,
>can you please provide a description and steps to reproduce

Please check [CLOUDSTACK-9606](https://issues.apache.org/jira/browse/CLOUDSTACK-9606).

**Comment by priyankparihar** [comment]
Hi @rajesh-battala, 
Changes are included. Thanks for your careful syntax observation.

**Comment by borisstoyanov** [comment]
@priyankparihar can you please add marvin tests and rebase against 4.9 so we could trigger smoke tests and merge it.

**Comment by priyankparihar** [comment]
@borisstoyanov please trigger smoke test.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by priyankparihar** [comment]
@borisstoyanov and @syed Please take a look. Test case is added. If now everything looks ok then please provide your LGTM.

**Comment by borisstoyanov** [comment]
@priyankparihar thank you for adding this test. Have you run it? Since this is not part of the smoketestd can you provide any results.

**Comment by sarathkouk** [comment]
Here is the test result,
Test for checking automatic tag removal on released Public IP ... === TestName: test_25_CLOUDSTACK_9606| Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 350.389s

OK

**Comment by sarathkouk** [comment]
[results.txt](https://github.com/apache/cloudstack/files/841130/results.txt)

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 701
 Hypervisor xenserver
 NetworkType Advanced
 Passed=112
 Failed=0
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by rhtyd** [comment]
@priyankparihar fix the conflicts please

**Comment by rhtyd** [comment]
@priyankparihar can you fix the conflict?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Re-kicking travis by closing and re-opening the PR.

**Comment by rhtyd** [comment]
@priyankparihar can you fix the error in the new marvin test, see Travis's failed job for details:
```
 test_25_CLOUDSTACK_9606    | exceptions.NameError | 16.268  | test_tags    
```

**Comment by DaanHoogland** [comment]
@rhtyd @priyankparihar @syed @rajesh-battala @sarathkouk @borisstoyanov 
As this PR does not pass it's own test in travis and does not seem to be very critical, I suggest we move it to a future release.

**Comment by rhtyd** [comment]
Agree @DaanHoogland I'll move this to 4.12 milestone.

**Comment by rafaelweingartner** [comment]
I will move this to 5.0.0 due to the lack of activity

**Comment by andrijapanicsb** [comment]
Seems abandoned by the author and has been dragging for 4 years - closing this one. Kindly reopen in case this is still active.

**Pull request closed without merging**